diff --git a/pypeit/spec2dobj.py b/pypeit/spec2dobj.py
index 45e090518..885926c0d 100644
--- a/pypeit/spec2dobj.py
+++ b/pypeit/spec2dobj.py
@@ -107,7 +107,7 @@ class Spec2DObj(datamodel.DataContainer):
             `Spec2DObj`:
 
         """
-        hdul = io.fits_open(file)
+        hdul = io.fits_open(file, memmap=False)
         # Quick check on det
         if not np.any(['DET{:02d}'.format(det) in hdu.name for hdu in hdul]):
             msgs.error("Requested detector {} is not in this file - {}".format(det, file))
@@ -264,7 +264,7 @@ class AllSpec2DObj(object):
         # Instantiate
         slf = cls()
         # Open
-        hdul = io.fits_open(filename)
+        hdul = io.fits_open(filename, memmap=False)
         # Meta
         hkeys = list(hdul[0].header.keys())
         for key in hkeys:
